services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/api_client/_public.py

```diff
@@ -401,28 +403,44 @@ async def restart_containers(self, dynamic_sidecar_endpoint: AnyHttpUrl) -> None
 
 
 async def setup(app: FastAPI) -> None:
-    logger.debug("dynamic-sidecar api client setup")
-    app.state.dynamic_sidecar_api_client = DynamicSidecarClient(app)
+    with log_context(logger, logging.DEBUG, "dynamic-sidecar api client setup"):
+        app.state.dynamic_sidecar_api_clients: dict[str, DynamicSidecarClient] = {}
 
 
 async def shutdown(app: FastAPI) -> None:
-    logger.debug("dynamic-sidecar api client closing...")
-    client: Optional[DynamicSidecarClient]
-    if client := app.state.dynamic_sidecar_api_client:
-        await client._thin_client.close()  # pylint: disable=protected-access
+    with log_context(logger, logging.DEBUG, "dynamic-sidecar api client closing..."):
+        await logged_gather(
```

Comment:
this can raise. maybe care to put reraise=False?

Reply:
But I want this to re raise if something is wrong.

Reply:
Yes, putting it to false.